### PR TITLE
Replace AutomationViewBuilder class with AutomationViewBuilderFactory service

### PR DESCRIPTION
**Pull request by niklasnatter**
Providing a `AutomationViewBuilderFactory` service allows to overwrite it inside of a project and is consistent to the `ViewBuilderFactory` implementation in the `sulu/sulu` package.

### CODE REVIEW

File: Resources/config/services.xml [hunk 1]
```diff
@@ -47,10 +47,13 @@
             <tag name="sulu_admin.list_metadata_loader" priority="1024"/>
         </service>
 
-        <service id="sulu_automation.admin" class="Sulu\Bundle\AutomationBundle\Admin\AutomationAdmin">
+        <service id="sulu_automation.automation_view_builder_factory" class="Sulu\Bundle\AutomationBundle\Admin\View\AutomationViewBuilderFactory">
             <argument type="service" id="sulu_admin.view_builder_factory"/>
-            <argument type="service" id="sulu_core.webspace.webspace_manager"/>
-            <argument>%sulu_admin.name%</argument>
+        </service>
```
Comment (alexander-schranz) [inline Resources/config/services.xml:40]: Think also an alias for this service should exist:

```suggestion
        </service>

        <service id="Sulu\Bundle\AutomationBundle\Admin\View\AutomationViewBuilderFactoryInterface" alias="sulu_automation.automation_view_builder_factory"/>
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: Admin/View/AutomationViewBuilderFactory.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            ->setListKey(Task::RESOURCE_KEY)
+            ->setFormKey('task_details')
+            ->setTabTitle('sulu_automation.automation')
+            ->addToolbarActions([
+                new ToolbarAction('sulu_admin.add'),
+                new ToolbarAction('sulu_admin.delete'),
+            ])
+            ->addTabBadges([$taskCountBadge])
+            ->addListAdapters(['table'])
+            ->setTabOrder(4096)
```
Comment (alexander-schranz) [inline Admin/View/AutomationViewBuilderFactory.php]: not sure if I would predefine here any value

Comment (niklasnatter) [reply Admin/View/AutomationViewBuilderFactory.php]: you are right - makes sense not to do that 🙂

**Comment by tstauer** [comment]
This seems to break compatiblity with the ArticleBundle in versions lower than 2.3.0:
https://github.com/sulu/SuluArticleBundle/blob/2.2.5/Admin/ArticleAdmin.php#L298

I believe there should have been a bump in the composer.json of SuluAutomationBundle regarding the required version of sulu/sulu in this pull-request.

**Comment by alexander-schranz** [comment]
@tstauer If you update the AutomationBundle you also need to update in this case the ArticleBundle. As the AutomationBundle was not official released when ArticleBundle 2.2.5 was released, so the ArticleBundle requires an update if you want to use the official release version of the AutomationBundle.

**Pull request merged**